tests/how_to_test_systems.rs

```diff
@@ -0,0 +1,113 @@
+use std::collections::HashMap;
+
+use bevy::{ecs::system::CommandQueue, prelude::*};
+
+struct Enemy {
+    hit_points: u32,
+}
+
+struct CharacterTemplate {
+    hit_points: HashMap<&'static str, u32>,
+}
+
+fn despawn_dead_enemies(mut commands: Commands, enemies: Query<(Entity, &Enemy)>) {
+    for (entity, enemy) in enemies.iter() {
+        if enemy.hit_points == 0 {
+            commands.entity(entity).despawn_recursive();
+        }
+    }
+}
+
+fn hurt_enemies(mut enemies: Query<&mut Enemy>) {
+    for mut enemy in enemies.iter_mut() {
+        enemy.hit_points -= 1;
+    }
+}
+
+fn spawn_enemy(mut commands: Commands, character_template: Res<CharacterTemplate>) {
+    commands.spawn().insert(Enemy {
+        hit_points: *character_template.hit_points.get("enemy").unwrap(),
+    });
+}
+
+#[test]
+fn did_hurt_enemy() {
+    // Setup world and commands
+    let mut world = World::default();
+    let mut queue = CommandQueue::default();
+    let mut commands = Commands::new(&mut queue, &world);
+
+    // Setup stage with our two systems
+    let mut update_stage = SystemStage::parallel();
+    update_stage.add_system(hurt_enemies.system().before("death"));
+    update_stage.add_system(despawn_dead_enemies.system().label("death"));
+    let mut schedule = Schedule::default();
+    schedule.add_stage("update", update_stage);
```

Comment:
Why do we create a new schedule instead of just running the stage? We should run the stage directly, and have a separate test with multiple stages for showcasing the schedule stuff.

Reply:
Yes, if stages can be run directly without a schedule, then creating a schedule (just for showcasing the api) could be misleading and make people think that they always need to create a schedule.